### PR TITLE
Edited README, fixes #30

### LINKED ISSUE #30: Edit README to be more concise and ordered more chronologically
Currently the README is not far off 4000 words and from my own experience I found this, combined with the order in which topics are displayed/introduced a bit confusing when knowing where to start.

I am proposing to edit down the text a little (not to lose meaning but simply to make things more concise). As well as thinking about the order things appear, e.g. moving some of the longer video content which is about other topics such as phoenix lower down. It's not that this content isn't useful (hence not removing it), it's just that if you're trying to learn Elixir, you probably want to understand the basics of the language before you leap into phoenix.

I will make sure that people know that there is more content for them that may be of interest by making sure its included in the contents.

I think these changes could help improve people's learning experiences by helping them to learn the bits of elixir they need faster.

### PR DESCRIPTION
**Pull request by Cleop**
- Reordered so 'What' comes before 'Why' for people who may not know what Elixir is before wanting to know why they should be using it.
- Removes repeating content and rephrases/ condenses existing content
- Removes video screenshots to try to shorten the length of the README
- Moves videos into 2 sections: introductory videos and extra resources ie. starter content, nice to have if you have the time content.
- Adds extra links for key terms that not all beginners may be familiar with e.g. dynamic typed language, immutable data, state.
- Sections out how to install and how to write elixir into clearer sections for better navigation by users
- Changes 'Handy Tips' section to 'Commands' to be more semantic

### CODE REVIEW

**Comment by Cleop** [comment]
Update: 
 - Add links for keywords to explain: macros, closures, pattern matching, garbage collection
- Reordering/ general editing for readability

**Comment by Cleop** [comment]
- Fixes #39 with details on enumerable types.
- Adds link to codeschool resource #27

**Comment by Cleop** [comment]
- Adds Elixir School resource as proposed in #27

File: README.md [hunk 1]
```diff
@@ -1,141 +1,109 @@
 # Learn ![elixir logo](http://elixir-lang.org/images/logo/logo.png "Elixir Logo")
 
-Learn the Elixir _language_, designed to build dynamic, functional,
-scalable and maintainable web applications!
-
 ## Contents
-- [Why?](#why)
 - [What?](#what)
+- [Why?](#why)
```
Comment (nelsonic) [inline README.md]: FYI: we _always_ "_Start With **Why**_" ... 😉 
https://youtu.be/sioZd3AxmnE

Comment (Cleop) [inline README.md]: Why do user stories start with the what before the why?

Comment (nelsonic) [inline README.md]: @Cleop the "_Why_" question is to help understand the person's motivation:
+ Why do I want a particular feature?
+ Why am I feeling pain _not_ having some functionality?

Starting with "_What_" gives the (_potential_) "_solution_" `before` we've _understood_ the challenge/problem the person is facing.

Comment (Cleop) [inline README.md]: Thanks @nelsonic, I've made the change.

Comment (nelsonic) [inline README.md]: @Cleop sweet! 🍭

**Pull request merged**